rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,693 @@
+different spaces. We could then have changed the structure to
+
+```ts
+url: { absUrl: string } | { application: string; path?: string, space?: string };
+```
+
+But this had some caveats:
+- `space` is an xpack plugin, adding this `space` property in an oss part of the code is problematic
+- The `space` API is not callable from core or oss, meaning that we would have to 'forge' the url to this space anyway
+- this is really not generic. If another plugin was to alter the basepath in another way, we would have needed to add it another property 
+
+So even if the 'parsable absolute url' approach seems fragile, it still felt better than this alternative.
+
+## We could use plain string instead of an enum for `GlobalSearchResult.type`
+
+The current static enum used for type
+
+```ts
+enum SEARCH_TYPE {
+  // non-exhaustive
+  applications = 10,
+  dashboard = 20,
+  visualization = 30,
+  search = 40,
+}
+
+/** @public */
+type GlobalSearchResultType = keyof typeof SEARCH_TYPE;
+
+interface GlobalSearchResult {
+  // [...]
+  type: GlobalSearchResultType;
+}
+```
+
+has some limitations:
+- It forces the enum to be modified every time a new type is added
+- 3rd party plugins cannot introduce new types
+
+We could change the API to accept plain strings for `GlobalSearchResult.type`. However, atm this enum approach 
+is needed as the ordinal values of the entries is used in results sorting. Changing to plain strings forces to find 
+an alternative sorting logic.
+
+## triggered searches could be cancelable
+
+In current specifications, once a search has been triggered, it's not possible to cancel it (see [Search cancellation](#search-cancellation))
+
+Main drawback of this decision is that if a `find` API consumer is going to perform multiple `find` requests and 
+only uses the last one (For example in the UI, a debounced `find` triggered on end-user keypress: On every new keypress,
+a new search is performed and the previous, potentially still executing, `find` call will still be executed), it can't
+cancel the previous active search, resulting on potential useless http calls and resources consumption.
+
+We could add an optional cancellation signal or observable in `GlobalSearchOptions` and propagate it to result providers.
+That way `find` calls could be aborted.
+
+However result providers would still have to manually handles this signal to cancel any http call or 
+other asynchronous task that could be pending. 
+
+Notes:
+- As this can be implemented with an additional option, this would be a non-breaking change and so could be done at 
+  a later time (even if it means that existing result provider would not handles this option until they implements it).
+
+## The GlobalSearch service could be provided as a plugin instead of a core service
+
+The GlobalSearch API could be provided and exposed from a plugin instead of `core`.
+
+Pros:
+- Less `core` API exposure
```

Comment:
If we put this inside core, is this the first example of a plugin in core that isn't actually owned by the platform team?  So far we've been holding firm that every plugin has an explicit owner.  Do you think platform should/would take this over, do you think we should support multiple team owners of core code, or do you think this is another "Pro", to put it in a separate plugin?

Reply:
> If we put this inside core, is this the first example of a plugin in core that isn't actually owned by the platform team

If the GS service is put inside core, it will be a core service/API. If put in a plugin, it will be a plugin service/API. I think either way, owners of this API will be the platform team. Not sure to understand exactly what you mean by `a plugin in core`

> Do you think platform should/would take this over, do you think we should support multiple team owners of core code

Owner of the GS API is the platform team
Owner of the GS result providers required for the v1 of https://github.com/elastic/kibana/issues/57576 is still unclear TBH. ATM it's likely that the platform team will perform the development. 

The multiple teams question is a very valid one, I'll take the recent core-ui PR for the new nav as an example (it's in core, but it feels like core-ui should become the new owner/co-owner) However this is way outside the scope of this RFC imho.